File: DisplaFrontend/src/app/model/comprobanteProveedor.ts
import { Gasto } from "./Gasto"
import { Proveedor } from "./Proveedor"
import { TarjetaCredito } from "./tarjetaCredito"
import { TipoComprobante } from "./tipoComprobante"

export interface ComprobanteProveedor {
    Id: number,
    Fecha: Date,
    FechaImputacion: Date,
    IdProveedor: number,
    IdGasto: number,
    Sucursal: number,
    Numero: number,
    Clase: string,
    Monto: number,
    IdTipoComprobante: number,
    Neto: number,
    Iva: number,
    ConceptosNoGravados: number,
    RetencionGanancias: number,
    RetencionIva: number,
    TasaIva: number,
    SobreTasaIva: number,
    MontoAdeudado: number,
    FechaBorrado: Date,
    MotivoBorrado: string,
    MesTarjeta: number,
    AnioTarjeta: number,
    IdTarjeta: number,
    PendienteDePago: boolean,
    Pib: number,
    CotizacionDolar: number,
    MontoDolar: number,
    GastoTransferencia: number,
    GastoBanco: number,
    MultiTasasIva: boolean,
    Tseh: number,
    NombreProveedor: string,

    IdGastoNavigation: Gasto,
    IdProveedorNavigation: Proveedor,
    IdTarjetaNavigation: TarjetaCredito,
    IdTipoComprobanteNavigation: TipoComprobante,
}

File: DisplaBackend/DisplaBackend/Models/ComprobanteProveedor.cs
using System;
using System.Collections.Generic;

namespace DisplaBackend.Models
{
    public partial class ComprobanteProveedor
    {
        public ComprobanteProveedor()
        {
            ComprobanteImportacion = new HashSet<ComprobanteImportacion>();
            ComprobanteIva = new HashSet<ComprobanteIva>();
        }

        public int Id { get; set; }
        public DateTime Fecha { get; set; }
        public DateTime FechaImputacion { get; set; }
        public int? IdProveedor { get; set; }
        public int IdGasto { get; set; }
        public int Sucursal { get; set; }
        public int Numero { get; set; }
        public string Clase { get; set; }
        public decimal? Monto { get; set; }
        public int IdTipoComprobante { get; set; }
        public decimal? SobreTasaIva { get; set; }
        public decimal? MontoAdeudado { get; set; }
        public DateTime? FechaBorrado { get; set; }
        public string MotivoBorrado { get; set; }
        public int? MesTarjeta { get; set; }
        public int? AnioTarjeta { get; set; }
        public int? IdTarjeta { get; set; }
        public bool? PendienteDePago { get; set; }
        public decimal? CotizacionDolar { get; set; }
        public decimal? MontoDolar { get; set; }
        public decimal? GastoTransferencia { get; set; }
        public decimal? GastoBanco { get; set; }
        public decimal? ConceptosNoGravados { get; set; }
        public decimal? RetencionGanancias { get; set; }
        public decimal? RetencionIva { get; set; }
        public decimal? Pib { get; set; }
        public decimal? Tseh { get; set; }

        public virtual Gasto IdGastoNavigation { get; set; }
        public virtual Proveedor IdProveedorNavigation { get; set; }
        public virtual TarjetaCredito IdTarjetaNavigation { get; set; }
        public virtual TipoComprobante IdTipoComprobanteNavigation { get; set; }
        public virtual ICollection<ComprobanteImportacion> ComprobanteImportacion { get; set; }
        public virtual ICollection<ComprobanteIva> ComprobanteIva { get; set; }
    }
}
